### PR TITLE
Integrate CodeQL with GitHub Actions for vulnerable analysis

### PR DESCRIPTION
**Pull request by nalinigans**
Integrate CodeQL with GitHub Actions for static vulnerability scanning of C++ and Java sources. CodeQL does not seem to handle path/path-ignore filters for C++, so for now, we have to manually dismiss alerts for htslib, rapidjson, etc. I will look into building htslib/rapidjson/tiledb out of the cloned GenomicsDB folder and see if that helps later.

### CODE REVIEW

**Comment by nalinigans** [comment]
@mlathara, let me know if this is useful at all.

File: .github/resources/codeql-config.yml [hunk 1]
```diff
@@ -0,0 +1,6 @@
+name: "CodeQL configuration"
+
+paths-ignore:
```
Comment (nalinigans) [inline .github/resources/codeql-config.yml:3]: paths-ignore does not work for C++ and Java sources as yet - https://docs.github.com/en/code-security/code-scanning/automatically-scanning-your-code-for-vulnerabilities-and-errors/configuring-code-scanning#specifying-directories-to-scan

**Review by mlathara — APPROVED** [review]
Cool. So this is scanning our code, and everything we've pulled in under `dependencies`. There used to be something that would scan the pom and alert us about dependency vulnerabilities there as well right? I wonder what happened to that...

**Comment by nalinigans** [comment]
> There used to be something that would scan the pom and alert us about dependency vulnerabilities there as well right? I wonder what happened to that...
That is suppossedly done by the [Depandabots](https://github.com/GenomicsDB/GenomicsDB/security/dependabot), but yes I have not see any alerts in awhile.

**Comment by nalinigans** [comment]
> So this is scanning our code, and everything we've pulled in under `dependencies`. 

Yes for now - which may not be a bad thing as TileDB/RapidJSON/htslib all get scanned.

**Pull request merged**